crates/biome_analyze/CONTRIBUTING.md

```diff
@@ -681,3 +681,57 @@ impl Rule for ExampleRule {
     }
 }
 ```
+
+### Semantic Model
+
+The semantic model provides information about the references of a binding (variable) within a program, indicating if it is written (e.g., `const a = 4`), read (e.g., `const b = a`, where `a` is read), or exported.
+
+
+#### Using the Semantic<> in rule
+
+We've a `for` loop and we need to identify the `i` variable usage:
```

Comment:
We should explain **where** we should identify its usage. Inside the loop?

Reply:
I didn't consider the scope in this example
So i want to show how to identify its usage in whole `for` loop

Reply:
How about

We have a `for` loop that creates an index `i`, and we need to identify where this index is used inside the body of the loop